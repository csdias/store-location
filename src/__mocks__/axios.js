const axiosMock = jest.genMockFromModule('axios');

let mockResponse = {
    data: { "shops": [
        { "location": "Portland", "address": "123 Portland Av"},
        { "location": "Astoria", "address": "456 Astoria St"},
        { "location": "", "address": ""}
    ]}
}

axiosMock.get.mockImplementation(req);

function req(){
    return new Promise(function(resolve){
        axiosMock.delayTimer = setTimeout(function(){
            resolve(mockResponse);
        }, 100);

    });
}

module.exports = axiosMock; 
