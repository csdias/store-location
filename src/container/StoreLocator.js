import React, { Component, useState, useEffect } from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import MapChooser from '../MapChooser';
import Header from '../component/Header';
import Map from '../component/Map';
import Button from '../component/Button';

const StoreLocatorFunction = () => {

    const [state, setState] = useState({ currentMap: 'none.png', shops: [] });

    useEffect(() => {

        async function fetchData(){
            const response =  await axios.get('http://localhost:3000/data/shops.json');

            setState({ shops: response.data.shops });
        }

        fetchData()

        return () => {};

    }, []);

    const chooseMap = (e) => {

        setState({ currentMap: MapChooser(e.target.value)});

    }

    const buttons = state.shops.map((shop, idx) => {  return <Button key={idx} handleClick={ chooseMap } location={shop.location} />})

    return ( <><Header/>{buttons} <Map imageName={state.currentMap} location={state.currentMap} /></> )

}


class StoreLocatorClass extends Component {

    constructor(props){

        super(props);

        this.state = {
            currentMap: 'none.png',
            shops: []
        };

        this.chooseMap = this.chooseMap.bind(this);

    }

    async componentDidMount(){

        let response = await axios.get('http://localhost:3000/data/shops.json');
        this.setState({ shops: response.data.shops });
        
    }

    chooseMap(e) {
        this.setState({ currentMap: MapChooser(e.target.value)});
    }

    teste(){
        alert('teste');
    }
    
    render(){

        //this.teste();

        let buttons = this.state.shops.map((shop, idx) => { 
            return <Button handleClick={this.chooseMap} key={idx} location={shop.location} />
        });

        return (
            <>
            <Header /><div>{buttons}</div><Map imageName={this.state.currentMap} location={this.state.currentMap}/>
            </>
        )

    }

}

export { StoreLocatorClass, StoreLocatorFunction };
//export default StoreLocator;