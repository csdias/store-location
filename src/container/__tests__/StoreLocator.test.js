import React from 'react';
import {shallow, mount} from 'enzyme';
import StoreLocator from '../StoreLocator';
import axios from 'axios';
import renderer from 'react-test-renderer';

describe('ServiceLocator', function(){

    let mountedStoreLocator;

    beforeEach(() => {
        mountedStoreLocator = shallow(<StoreLocator />);
    });

    it('Should render without crash', function(){
        mountedStoreLocator = shallow(<StoreLocator />);
    }); 

    it('Should render a header', function(){
        let headers = mountedStoreLocator.find('Header');
        expect(headers.length).toBe(1);
    });

    it('Should render 0 buttons', function(){
        let buttons = mountedStoreLocator.find('Button');
        expect(buttons.length).toBe(0);
    });

    it('Should render a map', function(){
        let map = mountedStoreLocator.find('Map');
        expect(map.length).toBe(1);

    });

    //practising
    it('Should render properly', () => {

        const tree = renderer.create(<StoreLocator />).toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('Should make a call to axios', () => {


        

    });

    it('renders correctly', function(){

        let tree = renderer.create(<StoreLocator />).toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('Calls axios.get from componentDidMount', function(){

        return mountedStoreLocator.instance().componentDidMount().then(function(){
            expect(axios.get).toHaveBeenCalled();
        });

    })

    it('Calls axios.get from componentDidMount with corret url', function(){

        return mountedStoreLocator.instance().componentDidMount().then(function(){
            expect(axios.get).toHaveBeenCalledWith('http://localhost:3001/data/shops.json');
        });

    }); 

    it('update the state with api data', function(){
        return mountedStoreLocator.instance().componentDidMount().then(function(){
            expect(mountedStoreLocator.state()).toHaveProperty('shops', 
                [
                    { location: 'Portland', address: '123 Portland Av'},
                    { location: 'Astoria', address: '456 Astoria St'},
                    { location: '', address: ''}
                ]            
            );
        });

    });

})

describe('ChooseMap', function(){

    it('Updates the state using the location passed to it', function(){

        let mountedStoreLocator = shallow(<StoreLocator />);
        let mockEvent = { target: { value: 'testland'} };
        mountedStoreLocator.instance().chooseMap(mockEvent);
        expect(mountedStoreLocator.instance().state.currentMap).toBe('testland.png');
        
    }) 

})

