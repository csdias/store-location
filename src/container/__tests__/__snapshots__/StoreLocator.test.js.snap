// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceLocator Should render properly 1`] = `
Array [
  <div
    className="Header"
  >
    <img
      alt="logo"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>,
  <div
    className="MapBox"
  >
    <img
      alt="none.png"
      src="images/none.png"
    />
  </div>,
]
`;

exports[`ServiceLocator renders correctly 1`] = `
Array [
  <div
    className="Header"
  >
    <img
      alt="logo"
      src="images/wired-brain-coffee-logo.png"
    />
  </div>,
  <div
    className="MapBox"
  >
    <img
      alt="none.png"
      src="images/none.png"
    />
  </div>,
]
`;
